python/cuml/cuml/naive_bayes/naive_bayes.py

```diff
@@ -802,9 +853,20 @@ def _partial_fit(
         if scipy.sparse.isspmatrix(X) or cupyx.scipy.sparse.isspmatrix(X):
             X = _convert_x_sparse(X)
         else:
-            X = input_to_cupy_array(
-                X, order="K", check_dtype=[cp.float32, cp.float64, cp.int32]
-            ).array
+            try:
+                X = input_to_cupy_array(
+                    X,
+                    order="K",
+                    check_dtype=[cp.float32, cp.float64, cp.int32],
+                ).array
+            except Exception as e:
```

Comment:
This helped get exceptions that pointed to the actual source of the problem, might be able to remove them again once we are done debugging. But unsure if there is any downside to having it here, the benefit is that users also get better error messages.

Same for the stream sync below